src/ai/backend/manager/models/vfolder.py

```diff
@@ -1587,6 +1548,74 @@ class Meta:
     items = graphene.List(VirtualFolder, required=True)
 
 
+async def delete_vfolders(
+    requested_vfolders: Sequence[VFolderDeletionInfo],
+    *,
+    storage_manager: StorageSessionManager,
+    db: ExtendedAsyncSAEngine,
+    reporter: ProgressReporter | None = None,
+) -> None:
+    async def _task(reporter: ProgressReporter | None):
+        row_deletion_infos: list[VFolderDeletionInfo] = []
+        failed_deletion: list[tuple[VFolderDeletionInfo, str]] = []
+        for vfolder_info in requested_vfolders:
+            folder_id, host_name = vfolder_info
+            proxy_name, volume_name = storage_manager.split_host(host_name)
+            log_type = LogType.INFO
+            try:
+                async with storage_manager.request(
+                    proxy_name,
+                    "POST",
+                    "folder/delete",
+                    json={
+                        "volume": volume_name,
+                        "vfid": str(folder_id),
+                    },
+                ) as (_, resp):
+                    pass
+            except (VFolderOperationFailed, InvalidAPIParameters) as e:
+                if e.status == 404:
+                    row_deletion_infos.append(vfolder_info)
+                    progress_msg = (
+                        f"VFolder not found, transit status to DELETE_COMPLETE (id: {folder_id})"
```

Comment:
Storage proxy raising 404 means it did not manage to find VFolder's corresponding physical path from its storage backend, not VFolder itself. This message needs to be rewrited so that it can represent situation more accurately.